File: Server/src/main/java/jk_5/nailed/server/command/CommandZone.java
package jk_5.nailed.server.command;

import com.google.common.base.Joiner;
import jk_5.nailed.api.map.Map;
import jk_5.nailed.api.player.Player;
import jk_5.nailed.api.zone.IZone;

import java.util.Set;

/**
 * No description given
 *
 * @author jk-5
 */
public class CommandZone extends NailedCommand {

    public CommandZone() {
        super("zone");
    }

    @Override
    public void processCommandPlayer(Player sender, Map map, String[] args) {
        Set<IZone> zones = map.getZoneManager().getZones(sender);
        sender.sendChat(Joiner.on(", ").join(zones));
    }
}

File: Server/src/main/scala/jk_5/nailed/server/command/CommandZone.scala
package jk_5.nailed.server.command

import jk_5.nailed.api.player.Player
import jk_5.nailed.api.map.Map
import scala.collection.JavaConversions._

/**
 * No description given
 *
 * @author jk-5
 */
object CommandZone extends ScalaCommand {

  val name = "zone"
  val usage = "/zone - Displays the zone you are currently in"

  override def processCommandPlayer(sender: Player, map: Map, args: Array[String]){
    val zones = map.getZoneManager.getZones(sender)
    sender.sendChat(zones.mkString(", "))
  }
}
